src/samplers/sampler.jl

```diff
@@ -32,7 +33,7 @@ assume(spl::Void, dist::Distribution, vn::VarName, vi::VarInfo) = begin
   if haskey(vi, vn)
     r = vi[vn]
   else
-    r = rand(dist)
+    r = init(dist)
```

Comment:
I'm a bit unsure if it is necessary when implementing this: the place this `Init` is need is the `model(sampler=Init(alg))` in `hmc.jl` and `gibbs.jl`. For both (even in `gibbs.jl`), the values initialized by `model()` seems to be never used by PG, because PG will always clear (i.e. set values to `NULL`) in the beginning of each step.